### PR TITLE
[14.0][MIG] delivery_postlogistics to 14.0

### PR DESCRIPTION
**Pull request by hailangvn**

### CODE REVIEW

File: delivery_postlogistics/views/delivery.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                        </group>
+                        <group string="ZPL Patch">
+                            <field name="zpl_patch_string" />
+                        </group>
+                    </group>
+                </page>
+            </page>
+        </field>
+    </record>
+    <!-- <record
```
Comment (leemannd) [inline delivery_postlogistics/views/delivery.xml]: Missing migration or to be removed?

File: delivery_postlogistics/models/stock_quant_package.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        if len(order) > 1:
+            raise exceptions.Warning(
+                _(
+                    "The cash on delivery amount must be manually specified "
+                    "on the packages when a package contains products "
+                    "from different sales orders."
+                )
+            )
+
+        # As history_ids column is removed, and stock.quant is not directly
```
Comment (leemannd) [inline delivery_postlogistics/models/stock_quant_package.py]: Missing migration of code or to be removed?

File: delivery_postlogistics/views/delivery.xml (hunk 1 above)
Comment (hailangvn) [reply delivery_postlogistics/views/delivery.xml]: I will remove this code in next update.

File: delivery_postlogistics/models/stock_quant_package.py (hunk 2 above)
Comment (hailangvn) [reply delivery_postlogistics/models/stock_quant_package.py]: There is comment that we may need a feature to have multiple packages for single delivery order. I am looking for suggestion.

**Comment by hailangvn** [comment]
Good day, this PR has been updated with improvements from 13.0.

**Comment by mmequignon** [comment]
runbot is :tomato:

**Comment by nilshamerlinck** [comment]
Impacted by https://github.com/OCA/maintainer-quality-tools/pull/697

**Comment by nilshamerlinck** [comment]
Certificate issue mentioned above has been fixed. We are now looking into the other issue that prevents runbot from being green.

**Comment by hailangvn** [comment]
All is green now. :)

File: delivery_postlogistics/__manifest__.py [hunk 3]
```diff
@@ -3,15 +3,21 @@
...
     "name": "PostLogistics Shipping - “Barcode” web service",
     "summary": "Print PostLogistics shipping labels using the Barcode web service",
-    "version": "13.0.1.2.0",
+    "version": "14.0.1.0.0",
     "author": "Camptocamp,Odoo Community Association (OCA)",
     "maintainer": "Camptocamp",
     "license": "AGPL-3",
     "category": "Delivery",
     "complexity": "normal",
+    "external_dependencies": {
```
Comment (simahawk) [inline delivery_postlogistics/__manifest__.py]: this is wrong. It's not a fucntional dependency. Such dep goes to test-requirements.txt at repo level.

File: delivery_postlogistics/models/stock_quant_package.py [hunk 4]
```diff
@@ -12,17 +12,20 @@ class StockQuantPackage(models.Model):
...
+        string="Packaging's Carrier",
+    )
 
     @api.returns("stock.picking")
     def _get_origin_pickings(self):
         self.ensure_one()
         move_line_model = self.env["stock.move.line"]
-        move_line = move_line_model.search([("package_id", "=", self.id)])
-        return move_line.mapped("picking_id")
+        move_line = move_line_model.search([("result_package_id", "=", self.id)])
```
Comment (simahawk) [inline delivery_postlogistics/models/stock_quant_package.py]: This change seems wrong. The method says "_get_origin" but here you look for destination packages now :thinking:

**Review by simahawk — CHANGES_REQUESTED** [review]

File: delivery_postlogistics/models/stock_picking.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+            raise exceptions.Warning(
+                _(
+                    "The cash on delivery amount must be manually specified "
+                    "on the packages when a package contains products "
+                    "from different sales orders."
+                )
+            )
+        order_moves = order.picking_ids
+        # check if the package delivers the whole sales order
+        if len(order_moves) > 1:
```
Comment (sebalix) [inline delivery_postlogistics/models/stock_picking.py]:
```suggestion
        # check if the package delivers the whole sales order
        if len(order.picking_ids) > 1:
```

File: delivery_postlogistics/models/stock_quant_package.py [hunk 6]
```diff
@@ -12,17 +12,20 @@ class StockQuantPackage(models.Model):
...
+        string="Packaging's Carrier",
+    )
 
     @api.returns("stock.picking")
     def _get_origin_pickings(self):
         self.ensure_one()
         move_line_model = self.env["stock.move.line"]
         move_line = move_line_model.search([("package_id", "=", self.id)])
-        return move_line.mapped("picking_id")
+        return move_line.mapped("picking_id.sale_id.picking_ids")
```
Comment (sebalix) [inline delivery_postlogistics/models/stock_quant_package.py:25]: Question: is this change part of the migration or a bug you spotted? If it's a bug, better to put such change in a separate commit so they can be backported easily on previous versions.

File: delivery_postlogistics/models/stock_quant_package.py [hunk 7]
```diff
@@ -59,16 +62,4 @@ def postlogistics_cod_amount(self):
...
-        # check if the package delivers the whole sales order
-        if order_moves != package_moves:
-            raise exceptions.Warning(
-                _(
-                    "The cash on delivery amount must be manually specified "
-                    "on the packages when a sales order is delivered "
-                    "in several packages."
-                )
-            )
-
```
Comment (sebalix) [inline delivery_postlogistics/models/stock_quant_package.py]: What is the reason to remove this?

**Review by sebalix — COMMENTED** [review]
Some questions

File: delivery_postlogistics/models/stock_quant_package.py (hunk 7 above)
Comment (sebalix) [reply delivery_postlogistics/models/stock_quant_package.py]: <s>Comment outdated, resolved!</s> I think my comment is still relevant, the migration commit is removing this piece of code.

Comment (hailangvn) [reply delivery_postlogistics/models/stock_quant_package.py]: Above fix and this one are split and combined into new commit.

**Comment by hailangvn** [comment]
Thanks @sebalix for your review. This PR has been updated accordingly.

**Comment by simahawk** [comment]
@sebalix good for you?

**Review by sebalix — APPROVED** [review]
Yes, LG
@mmequignon will probably fix the tests/cassettes as some changes occurred on SwissPost API, let's merge this already.

**Review by simahawk — APPROVED** [review]
